### DISCUSSION TITLE
RPi Undervoltage

### DISCUSSION THREAD

**Discussion by EvoSems** [awawa-dev/HyperHDR#862] [General]
Hi guys, recently came back to the project after a few years away with no ambilight. 

successfully have a new setup with parallel hyperspi reverse s2 mini all working great.

my only issue is the rpi under voltage warning. I am running the pi with the same power supply as the leds all works great with the correct grounding. but how do I run the pi at the correct voltage? the psu is 5v/30a

if I use the pi charger then I have the grounding issue and flickering.  and im not sure exactly how to ground the pi charger separately 

assuming the pi needs 5.1v?

cheers all.

happy to be back!

**Comment by EvoSems** [comment]
![IMG_2553](https://github.com/awawa-dev/HyperHDR/assets/65807296/f0e05db1-5115-48ed-a8f8-c8eb7cb05a89)

**Comment by awawa-dev** [comment]
Hi

> successfully have a new setup with parallel hyperspi reverse s2 mini all working great.
my only issue is the rpi under voltage warning. I am running the pi with the same power supply as the leds all works great with the correct grounding. but how do I run the pi at the correct voltage? the psu is 5v/30a

Under-voltage does not always cause serious symptoms, but it is an undesirable condition, it is difficult to say whether it will work stably all the time. It probably depends on the scale of the problem but the RPi sensors will not report such details.

> if I use the pi charger then I have the grounding issue and flickering. and im not sure exactly how to ground the pi charger separately

Do you have ground connected from the LED strip to your esp32 s2 mini?

**Reply by EvoSems** [reply]
Hi Awawa. thanks for getting in touch. I
 remember long ago talking about the Ezcap320 with you. (currently trying to revive my 320 as it won't turn on after a few years in the drawer) 

Yes everything is grounded correctly between psu - pi - esp32 s2 - level shifter - led

ok if its not going to be an issue that's fine. (it was just the hot emoji symbol that worried me lol)

<img width="1321" alt="Screenshot 2024-05-22 at 18 10 00" src="https://github.com/awawa-dev/HyperHDR/assets/65807296/decd395f-4369-41b6-b8e9-d558f277a127">

loving hyperspi with parallel mode btw!

cheers

**Reply by awawa-dev** [reply]
Sometimes too much common ground causes real problems ;) Avoid loops, connect ground straight from the LED strip to the esp32 and nowhere else. I've encountered some really crazy problems in the past, e.g.: https://github.com/awawa-dev/HyperHDR/discussions/191#discussioncomment-2162246 everything was connected correctly with the level shifter, but the ground was divided in several places.

**Reply by awawa-dev** [reply]
sorry, connect ground from LED strip not to esp32 only to the level shifter in your case

**Reply by EvoSems** [reply]
Awesome thanks. I might get some time tomorrow to get the tv off the wall and tinker!

for anyone wanting 'CEC Control' and wants to keep 'Dolby Vision'
 I purchased a UGREEN Hdmi splitter which works perfectly with the Apple TV switching correctly between formats SDR/DV/HDR
https://www.amazon.com.au/dp/B0B129NNV1?psc=1&ref=ppx_yo2ov_dt_b_product_details

**Reply by EvoSems** [reply]
would this be the way?

![](https://user-images.githubusercontent.com/81187444/155578823-314c29a4-6271-4cd9-9e41-8a3e22ececf0.jpg)

**Reply by awawa-dev** [reply]
This diagram was prepared by a user who previously had problems in the attached topic. This one worked for him. I would simplify it, connect the level shifter with the LED strip, without a direct connection to the power supply. The connection from esp32 to the power supply i rather not neccesery: the data signal is prepared in the final phase and sent to the LED strip[ by the level shifter/not esp32.

**Reply by computergreek** [reply]
> Awesome thanks. I might get some time tomorrow to get the tv off the wall and tinker!
> 
> for anyone wanting 'CEC Control' and wants to keep 'Dolby Vision' I purchased a UGREEN Hdmi splitter which works perfectly with the Apple TV switching correctly between formats SDR/DV/HDR https://www.amazon.com.au/dp/B0B129NNV1?psc=1&ref=ppx_yo2ov_dt_b_product_details

i couldn't find that in stock anywhere in the states to order, so i bought what i thought was the closest thing from UGREEN:
https://www.amazon.com/dp/B09Y1594V7/ 

with this one, what i've found is that i can get CEC to work along with Dolby Vision/Atmos, but.......whenever i'm watching something passing through Dolby Vision, the colors being picked up by the MS2130 capture card are all really pink. did you have that problem too?

**Comment by EvoSems** [comment]
All Fixed! I have the Pi running from the Official Charger now with no undervoltage and no ground issues. (in parallel esp32-s2 with the level shifter.) Very happy! many thanks for your help (again)
